hr-rfe1-dagobah/fe-capstone
# Testing Setup

References [ticket #185](https://trello.com/c/mDyfQY2C/185-investigate-testing-set-up-a-mock-server-for-testing)

- Creates separate Jest config files for testing react or the server
- To test react: `npm test`
- To test server: `npm run test-server`
- Create test files next to the component they correspond to Component.jsx becomes Component.test.jsx
- Please do not commit failing tests
- Creates a mock server with msw to handle API get requests (other request types can be added)

Helpful Links:
[Jest Expect Extensions](https://jestjs.io/docs/en/expect)
[Testing Library - how to find / select elements on page](https://testing-library.com/docs/queries/about)
